### ISSUE TITLE
Feature Flag Haiku OS Support To Remove C++ Dependency If Not Used By Library Consumer 

### ISSUE THREAD

**Issue by LuisOsta** [strawlab/iana-time-zone#89]
Some time ago `iana-time-zone` [added support](https://github.com/strawlab/iana-time-zone/tree/main/haiku) for HaikuOS. Which ended up requiring a dependency on C++ in order to work. Which means that all end consumers of `iana-time-zone` are also dependent on C++.

Would it be possible to feature gate the Haiku OS support that only consumers that actually need to support Haiku OS end up with the C++ dependency? I am dependent on `iana-time-zone` through the `chrono`[ cargo package](https://github.com/chronotope/chrono) and would love to be able to remove that C++ dependency but its not possible while continuing to use `chrono`.

For context we are currently working on a Self Sovereign Identity library in Rust and are evaluating options for how to move forward - https://github.com/knox-networks/ssi/issues/71. Would love to continue using `chrono` and `iana-time-zone` but let me know if feature flagging isn't something you can do.

**Comment by Kijewski** [comment]
Please see #88. The build script is not being executed unless the target platform is actually haiku, so even if you don't have a c++ compiler installed, it should still work for all other platforms.

**Comment by LuisOsta** [comment]
Yeah the build scripts are not being executed but the dependency to the haiku crate and hence C++ still remains. Would it not be possible to remove that dependency via feature flag?

Here's the relevant sample of the Cargo.lock:
```
[[package]]
name = "chrono"
version = "0.4.22"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "bfd4d1b31faaa3a89d7934dbded3111da0d2ef28e3ebccdb4f0179f5929d1ef1"
dependencies = [
 "iana-time-zone",
 "js-sys",
 "num-integer",
 "num-traits",
 "serde",
 "time 0.1.43",
 "wasm-bindgen",
 "winapi",
]
...

[[package]]
name = "iana-time-zone"
version = "0.1.51"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "f5a6ef98976b22b3b7f2f3a806f858cb862044cfa66805aa3ad84cb3d3b785ed"
dependencies = [
 "android_system_properties",
 "core-foundation-sys",
 "iana-time-zone-haiku",
 "js-sys",
 "wasm-bindgen",
 "winapi",
]

...
[[package]]
name = "iana-time-zone-haiku"
version = "0.1.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "0703ae284fc167426161c2e3f1da3ea71d94b21bedbcc9494e92b28e334e3dca"
dependencies = [
 "cxx",
 "cxx-build",
]
```

**Comment by lopopolo** [comment]
The haiku platform crate is not built unless building for a haiku target in the same way that `wasm-bindgen` is not built unless building for Wasm targets or `winapi` is not built unless building for Windows targets.

Can you dig in a bit more on the problem are you trying to solve with the feature gate? Is it that new dependencies appearing in the lockfile are undesirable? This is how cargo works: all dependencies for all platforms are stored in the lockfile since cargo does not make assumptions on what platforms you will target.

What extra costs are you perceiving to the haiku integration? These same costs are also there for Windows users that only build their projects on Linux irrespective of this crate's haiku platform support.
